docs/guide.rst

```diff
@@ -86,3 +86,141 @@ options::
 
 Go on and explore. Don't be shy and ask, ``--help`` is eager to
 answer all your questions ;-)
+
+
+Under The Hood
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now let's have a brief look under the hood. Similarly as with
+``git``, there is a special ``.fmf`` directory which marks root of
+the fmf metadata tree. Use the ``init`` command to initialize it::
+
+    tmt init
+
+Do not forget to include this special ``.fmf`` directory in your
+commits, it is essential for building the fmf tree structure which
+is created from all ``*.fmf`` files discovered under the fmf root.
+
+
+Plans
+------------------------------------------------------------------
+
+As we've seen above, in order to enable a simple test the
+following plan is just enough::
+
+    execute:
+        script: foo --version
+
+Store these two lines in a ``*.fmf`` file and that's it. Name and
+location of the file is completely up to you, plans are recognized
+by the ``execute`` key which is required. Once the newly created
+plan is submitted to the CI system test script will be executed.
+
+:ref:`/spec/plans` are used to enable testing and group relevant
+tests together. They describe how to :ref:`/spec/plans/discover`
+tests for execution, how to :ref:`/spec/plans/provision` the
+environment, how to :ref:`/spec/plans/prepare` it for testing, how
+to :ref:`/spec/plans/execute` tests, :ref:`/spec/plans/report`
+results and finally how to :ref:`/spec/plans/finish` the test job.
+
+Here's an example of a slightly more complex plan which changes
+the default provision method to container to speed up the testing
+process and ensures that an additional package is installed before
+the testing starts::
+
+    provision:
+        how: container
+        image: fedora:33
+    prepare:
+        how: install
+        package: wget
+    execute:
+        how: tmt
+        script: wget http://example.org/
+
+
+Tests
+------------------------------------------------------------------
+
+Very often testing is much more complex than running just a
+single shell script. There might be many scenarios covered by
+individual scripts. For these cases the ``discover`` step can
+be instructed to explore available tests from fmf metadata as
+well. The plan will look like this::
+
+    discover:
+        how: fmf
+    execute:
+        how: tmt
+
+Tests, identified by the required key ``test``, define attributes
+which are closely related to individual test cases such as the
+:ref:`/spec/tests/test` script, :ref:`/spec/tests/framework`,
+directory :ref:`/spec/tests/path` where the test should be
+executed, maximum test :ref:`/spec/tests/duration` or packages
+required to run the test. Here's an example of test metadata::
```

Comment:
I think it makes sense to add a link to `required` here:
:ref:`/spec/tests/require`

Reply:
Good point, added.